File: Single Number.cpp
// Leetcode Problem 136: Single Number
// C++ CODE
#include <vector>
#include <numeric>
#include <functional>

class Solution {
public:
    int singleNumber(std::vector<int>& nums) {
        int singleNum = 0;
        for (int num : nums) {
            singleNum ^= num;
        }
        return singleNum;
    }
};

File: Single Number.py
# Leetcode Problem 136: Single Number
# PYTHON CODE
from functools import reduce
from typing import List


class Solution:

    def singleNumber(self, nums: List[int]) -> int:
        single_num = 0
        for num in nums:
            single_num ^= num
        return single_num


# Alternative one-liner using functools.reduce
class SolutionFunctional:

    def singleNumber(self, nums: List[int]) -> int:
        return reduce(lambda x, y: x ^ y, nums)